docs/building-extensions/components/component-examples/ajaxdemo.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Ajax demo component
+---
+
+Ajaxdemo Component
+==================
+
+This is an example component which you can download from [com_ajaxdemo](https://github.com/joomla/manual-examples/tree/main/component-ajaxdemo).
+
+It demonstrates the use of Ajax within a Joomla component; 
+for further background see the section [Ajax and JsonResponse](../../../general-concepts/javascript/ajax.md).
+
+It can be easily adapted to demonstrate the use of [com_ajax](../../../general-concepts/javascript/com-ajax.md) for plugins and templates; 
+simply change the url in media/js/divide.js to point to com_ajax instead of com_ajaxdemo, and set the other required URL parameters.
+
+Once you have downloaded the source, zip up the com_ajaxdemo directory and install the component.
+
+Then go to `<your domain>/index.php/component/ajaxdemo` to run it on your Joomla instance.
+
+The component displays a form to capture two numbers A and B, and a button to calculate A/B. 
+The division is performed by an Ajax call to the server, and if B is zero then an exception is raised.
+
+Brief summaries of the main source files are provided below.
+
+## administrator/components/com_ajaxdemo/services/provider.php
+
+This is boilerplate code for a basic MVC component. For components this file is placed under /administrator in the Joomla filesystem.
+If you want to understand it fully then read the [Dependency Injection](../../../general-concepts/dependency-injection/index.md) section. 
+
+From this file Joomla instantiates default [Extension and Dispatcher classes](../../../general-concepts/extension-and-dispatcher/index.md), 
+and an MVC Factory class which creates Model, View and Controller classes for our component. 
+
+The default Dispatcher class examines the URL `task` parameter, and from it works out the Controller to instantiate. 
+The default Controller (when no `task` parameter is present) is the DisplayController.
+When the Divide button is pressed the JavaScript code sends an Ajax request using `task` set to "ajax.divide". 
+Based on this the default Dispatcher will instantiate this component's AjaxController and call its `divide` method. 
+
+## Display Controller (components/com_ajaxdemo/src/Controller/DisplayController.php)
```

Comment:
I do not know about others. But I realy would suggest you to not to use path in the headings.
It is hardly readable, it is complicated to understand the section subject, and probably will confuse search engine, and anchors will be crazy :wink: 

I would suggest something like:
```
## Component service provider
path: `administrator/components/com_ajaxdemo/services/provider.php`
```

```
## Display Controller
path: `components/com_ajaxdemo/src/Controller/DisplayController.php`
```

It not for this PR, but in general :wink:

Reply:
Yes, you're right. I was thinking myself that it looked pretty bad.

I'll fix for this PR while I'm working on it.